File: android/src/main/java/com/wortise/ads/react/RNWortisePackage.java
package com.wortise.ads.react;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.JavaScriptModule;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

public class RNWortisePackage implements ReactPackage {
    // Deprecated from RN 0.47
    public List<Class<? extends JavaScriptModule>> createJSModules() {
      return Collections.emptyList();
    }

    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
      return Arrays.<NativeModule>asList(
        new RNWortiseAdSettings    (reactContext),
        new RNWortiseAppOpen       (reactContext),
        new RNWortiseConsentManager(reactContext),
        new RNWortiseDataManager   (reactContext),
        new RNWortiseInterstitial  (reactContext),
        new RNWortiseRewarded      (reactContext),
        new RNWortiseSdk           (reactContext)
      );
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
      return Arrays.<ViewManager>asList(
        new RNWortiseBanner()
      );
    }
}

File: android/src/main/java/com/wortise/ads/react/RNWortisePackage.kt
package com.wortise.ads.react

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.JavaScriptModule
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

class RNWortisePackage : ReactPackage {

  override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> = listOf(
    RNWortiseAdSettings    (reactContext),
    RNWortiseAppOpen       (reactContext),
    RNWortiseConsentManager(reactContext),
    RNWortiseDataManager   (reactContext),
    RNWortiseInterstitial  (reactContext),
    RNWortiseRewarded      (reactContext),
    RNWortiseSdk           (reactContext)
  )

  override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> = listOf(
    RNWortiseBanner()
  )
}
